public/app/features/datasources/state/actions.ts

```diff
@@ -67,17 +73,23 @@ export const initDataSourceSettings = (
   };
 };
 
-export const testDataSource = (dataSourceName: string): ThunkResult<void> => {
+export const testDataSource = (
+  dataSourceName: string,
+  dependencies: TestDataSourceDependencies = {
```

Comment:
I'm not sure changing this API is required. If I understand correctly the issue was that in the tests the services were not instantiated. I think mocking the required services as you did here https://github.com/grafana/grafana/blob/ee93fdff592e6c186a206b194297c3378670e897/public/app/features/datasources/state/actions.test.ts#L21 and then applying @grafana/runtime mock as we do for example here: https://github.com/grafana/grafana/blob/cf2cc713933599e7646416a56a665282c9d9e3bb/public/app/plugins/datasource/grafana-azure-monitor-datasource/app_insights/app_insights_datasource.test.ts#L6 should do the job.

Let me know @shavonn if that's unclear!

Reply:
@hugohaggmark FYI ^^

Reply:
We did this pattern previously for InitDataSources so I think that's why Shavonn continued this way.

Reply:
Makes it superclear what dependencies a thunk needs imo.

Reply:
@dprokop ^^^

Reply:
@shavonn We have discussed above considerations with @hugohaggmark and agreed the implementation proposed by you is better than the one suggested by me. I have not noticed the defaults in the thunk definition at first, which led me to have second thoughts. Let's merge this as it is!

Reply:
Thanks! Was out the last couple of days.